HotelReservation/HotelReservationTests/ReservationSteps/StepsExecutorTests.cs

```diff
@@ -35,40 +33,50 @@ public void ShouldExecuteCreatedStep()
         }
 
         [Test]
-        public void ShouldExecuteRealStep()
+        public void ShouldExecuteRealStepAndReturnStepOutputs()
         {
             var reservationStartProcessStepDouble = A.Fake<ReservationStartProcess>();
-            A.CallTo(() => reservationStartProcessStepDouble.Execute(_stepInputList)).DoesNothing();
+            var stepOutputDouble = new StepOutput(true, "message");
+            var expectedStepOutputs = new List<StepOutput> { stepOutputDouble };
 
-            _subject.ExecuteSteps(new List<IReservationStep> { reservationStartProcessStepDouble }, _stepInputList);
+            A.CallTo(() => reservationStartProcessStepDouble.Execute(_stepInputList)).Returns(stepOutputDouble);
 
-            A.CallTo(() => reservationStartProcessStepDouble.Execute(_stepInputList)).MustHaveHappened();
+            var stepOutputs = _subject.ExecuteSteps(new List<IReservationStep> { reservationStartProcessStepDouble }, _stepInputList);
+
+            stepOutputs.Should().BeEquivalentTo(expectedStepOutputs);
         }
 
         [Test]
         public void ShouldExecuteThreeCreatedSteps()
         {
-            var StepDouble = A.Fake<IReservationStep>();
+            var stepDouble = A.Fake<IReservationStep>();
 
-            var providedListOfSteps = new List<IReservationStep> { StepDouble, StepDouble, StepDouble};
+            var providedListOfSteps = new List<IReservationStep> { stepDouble, stepDouble, stepDouble };
 
             _subject.ExecuteSteps(providedListOfSteps, _stepInputList);
 
-            A.CallTo(() => StepDouble.Execute(_stepInputList)).MustHaveHappened(3, Times.Exactly);
+            A.CallTo(() => stepDouble.Execute(_stepInputList)).MustHaveHappened(3, Times.Exactly);
         }
 
         [Test]
-        public void ShouldExecuteThreeRealSteps()
+        public void ShouldExecuteThreeRealStepsAndReturnStepOutputs()
```

Comment:
Testy do tej klasy powinny być oparte tylko i wyłącznie o IReservationStep. Nie ma tam nic specyficznego dla konkretnych implementacji tego interfejsu. Przerób proszę te testy, żeby testowały to, co naprawdę się dzieje:
* Odpalamy wszystkie stepy
* Każdemu przekazujemy input
* Zbieramy wszystkie rezultaty

Reply:
Fixed